wp-includes/class-wp-service-workers.php

```diff
@@ -63,6 +135,119 @@ public function init() {
+			'modulePathPrefix' => PWA_PLUGIN_URL . $workbox_dir,
+		);
+		$script .= sprintf( "workbox.setConfig( %s );\n", wp_json_encode( $options, 64 /* JSON_UNESCAPED_SLASHES */ ) );
+
+		/**
+		 * Filters whether navigation preload is enabled.
+		 *
+		 * The filtered value will be sent as the Service-Worker-Navigation-Preload header value if a truthy string.
+		 * This filter should be set to return false to disable navigation preload such as when a site is using
+		 * the app shell model.
+		 *
+		 * @param bool|string $navigation_preload Whether to use navigation preload.
+		 */
+		$navigation_preload = apply_filters( 'service_worker_navigation_preload', true ); // @todo This needs to vary between admin and backend.
+		if ( false !== $navigation_preload ) {
+			if ( is_string( $navigation_preload ) ) {
+				$script .= sprintf( "workbox.navigationPreload.enable( %s );\n", wp_json_encode( $navigation_preload ) );
+			} else {
+				$script .= "workbox.navigationPreload.enable();\n";
+			}
+		} else {
+			$script .= "/* Navigation preload disabled. */\n";
+		}
+
+		return $script;
+	}
+
+	/**
+	 * Add precaching for wp-admin and wp-includes .js and .css files.
+	 */
+	protected function precache_admin_assets() {
+
+		$admin_dir    = ABSPATH . 'wp-admin/';
+		$admin_images = list_files( $admin_dir . 'images/' );
+		$inc_images   = list_files( ABSPATH . WPINC . '/images/' );
+
+		$routes = array_merge(
+			$this->get_routes_from_file_list( $admin_images, 'wp-admin' ),
+			$this->get_routes_from_file_list( $inc_images, 'wp-includes' ),
+			$this->get_admin_routes_from_dependency_list( wp_scripts()->registered ),
+			$this->get_admin_routes_from_dependency_list( wp_styles()->registered )
+		);
+
+		if ( empty( $routes ) ) {
+			return;
+		}
+
+		$this->register_precached_routes( $routes );
+	}
+
+	/**
+	 * Get routes from dependency list.
+	 *
+	 * @param array $dependencies Array of _WP_Dependency objects.
+	 * @return array Array of routes.
+	 */
+	protected function get_admin_routes_from_dependency_list( $dependencies ) {
+		$routes = array();
+		foreach ( $dependencies as $handle => $params ) {
+
+			// Only precache scripts from wp-admin and wp-includes.
+			if ( false === strpos( $params->src, 'wp-admin' ) && false === strpos( $params->src, 'wp-includes' ) ) {
+				continue;
+			}
+			$revision = false === $params->ver ? get_bloginfo( 'version' ) : $params->ver;
+			$routes[] = array(
+				'url'      => $params->src,
+				'revision' => $revision,
+			);
+		}
+		return $routes;
+	}
+
+	/**
+	 * Get routes from file paths list.
+	 *
+	 * @param array  $list List of file paths.
+	 * @param string $folder Folder -- either 'wp-admin' or 'wp-includes'.
+	 * @return array List of routes.
+	 */
+	protected function get_routes_from_file_list( $list, $folder ) {
+		$routes = array();
+		foreach ( $list as $filename ) {
+			$ext = pathinfo( $filename, PATHINFO_EXTENSION );
+			if ( ! in_array( $ext, array( 'png', 'gif', 'svg' ), true ) ) {
```

Comment:
I was going to ask about JPEG but then I saw that there are no JPEGS in core. For reference, the static assets are (via `git ls-files wp-includes/ wp-admin/ | sed 's/.*\.//' | sort | uniq -c | sort -nr`):

```
  99 png
  62 gif
  51 css
  10 scss
   7 txt
   6 svg
   3 woff
   3 ttf
   3 eot
   1 xml
   1 crt
```

It doesn't make sense to include `crt`, `eot` , `xml`, or `scss`. But `woff` would be very useful. Also, if this is used for themes then it would be good to include the JPEG extension(s).

Reply:
Thinking that maybe instead of using `list_files` for getting the 3 `woff` files we could list these out manually?

Reply:
Added the three `.woff` files as a static list within 7132b74.